pkg/cmd/server/origin/master.go

```diff
@@ -380,6 +382,34 @@ func initAPIVersionRoute(root *restful.WebService, prefix string, versions ...st
 		Consumes(restful.MIME_JSON))
 }
 
+// initReadinessCheckRoute initializes an HTTP endpoint for readiness checking
+func initReadinessCheckRoute(root *restful.WebService, path string, readyFunc func() bool, client *osclient.Client) {
+	root.Route(root.GET(path).To(func(req *restful.Request, resp *restful.Response) {
+		var healthzResponse kapi.Status
+		err := client.Get().AbsPath("/healthz").Do().Into(healthzResponse)
```

Comment:
I don't think this adds much in retrospect.  Let's just do the ready check.  We can always come back later and jazz it up.